internal/client/client.go

```diff
@@ -545,14 +546,20 @@ func RetryableClient(cfg *config.Config) *http.Client {
 	return retryableClient.StandardClient()
 }
 
-func PooledTransport() *http.Transport {
+func PooledTransport(cfg *config.Config) *http.Transport {
+	var tlsConfig *tls.Config
+	if (cfg.TLSConfig != nil) && cfg.TLSConfig.InsecureSkipVerify {
```

Comment:
why does this assign only if skipping verification?

Reply:
I want to be backward compatible as much as possible. If it is not set (which will be 99% cases), it will be same as before. `TLSClientConfig` is still `nill`. So no change will be introduced.